mantidproject/mantid
# Add MDFrameValidator to Mantid

**Description of work.**

Add a MD workspace validator for checking that a MDWorkspace conforms to a particular frame (e.g. HKL, QLab, QSample...). I want to use this in another PR (#21552) to check that the MD workspace being past is in HKL, but this code can be reviewed separately of that work.

**To test:**

Not much to test that isn't covered by the unit tests. 
 - Code review
 - If you're keen, make a quick python algorithm ([see python course for how to](https://www.mantidproject.org/Workspace_Validators)) that uses the validator and check it works by passing an MD workspace with/without a particular frame.

*No associated issue*

Does this update require release notes?


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
